warehouse/validations/validate_test.go

```diff
@@ -50,25 +50,43 @@ func TestValidator(t *testing.T) {
 
 	t.Run("Object Storage", func(t *testing.T) {
 		t.Run("Non Datalakes", func(t *testing.T) {
-			v, err := validations.NewValidator(ctx, model.VerifyingObjectStorage, &backendconfig.DestinationT{
-				DestinationDefinition: backendconfig.DestinationDefinitionT{
-					Name: warehouseutils.POSTGRES,
+			testCases := []struct {
+				name                      string
+				cleanupObjectStorageFiles bool
+			}{
+				{
+					name:                      "check delete permissions",
+					cleanupObjectStorageFiles: true,
 				},
-				Config: map[string]interface{}{
-					"host":            pgResource.Host,
-					"port":            pgResource.Port,
-					"database":        pgResource.Database,
-					"user":            pgResource.User,
-					"password":        pgResource.Password,
-					"bucketProvider":  provider,
-					"bucketName":      minioResource.BucketName,
-					"accessKeyID":     minioResource.AccessKeyID,
-					"secretAccessKey": minioResource.AccessKeySecret,
-					"endPoint":        minioResource.Endpoint,
+				{
+					name:                      "check delete permissions",
+					cleanupObjectStorageFiles: false,
 				},
-			})
-			require.NoError(t, err)
-			require.NoError(t, v.Validate(ctx))
+			}
```

Comment:
Can we have different names for individual tests. Also keep it within `t.Run`.

Reply:
done